tasks/upload_processor.py

```diff
@@ -522,25 +484,23 @@ def _attempt_rewrite_raw_report_readable_error_case(
 
     def _rewrite_raw_report_readable(
         self,
-        processing_result: dict,
+        raw_report: ParsedRawReport,
```

Comment:
Seeing that the previous implementation had an `if raw_report`, should this be typed with optional?

Reply:
The check for `if raw_report` was moved to the caller instead. Though one of call sites goes through `ReportService.parse_raw_report_from_storage`, and I trust that its return type annotation is correct in this case.